src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -149,6 +149,21 @@
     /**
      * {@inheritDoc}
      */
+    @NonNull
+    @Override
+    public <C extends Credentials> List<C> getCredentials(@NonNull Class<C> type, @NonNull Item item,
+                                                          @Nullable Authentication authentication,
+                                                          @NonNull List<DomainRequirement> domainRequirements) {
+        if (item instanceof AbstractFolder) {
+            // credentials defined in the folder should be available in the context of the folder
+            return getCredentials(type, (ItemGroup) item, authentication, domainRequirements);
+        }
+        return super.getCredentials(type, item, authentication, domainRequirements);
```

Comment:
Maybe makes sense to merge results into a single list. Just in case super's behavior changes